license/src/com/untangle/app/license/LicenseManagerImpl.java

```diff
@@ -734,16 +747,21 @@ private synchronized void _downloadLicenses()
             if(parse.has("restricted")) {
                 boolean restrict = parse.getBoolean("restricted");
                 restricted = restrict;
+
+                // clear licenses if restricted, we'll only use licenses used in response
+                if (restricted) {
+                    this.settings.getLicenses().clear();
```

Comment:
can get licenses return null? is it possible this can happen in all instances (if our response from license server was good), and revocations can be removed? update NGFW-13744 if this is implemented here

Reply:
Do this in the context of NGFW-13744